File: hw1/city.py
class City:

    def __init__(self,size : float, name : str, population : int = 0) -> None:
        self._size : float = size
        self._name : str = name
        self._population : int = population

    def isNewCity(self) -> bool:
        return self.population == 0

    def density(self) -> str:
        if self._population < 100000:
            return "Town"
        elif self._population < 10000000:
            return "City"
        else: 
            return "MegaCity"

    @property
    def population(self):
        return self._population

    @property 
    def size(self):
        return self._size

    @property
    def name(self) -> str:
        return self._name

    @size.setter
    def landSize(self, area: float) -> None:
        if area < 0:
            raise ValueError('Error, Not Enough Land!')
        self._size = area

File: hw2/src/City.cpp
#include "City.h"
#include <stdexcept>

namespace city {
    City::City(float size, std::string name, int population) {
        c_size = size;
        c_name = name;
        c_population = population;
    }
    bool City::isNewCity() { return c_population == 0; }

    std::string City::density() {
        if(c_population < 100000) { return "Town"; }
        else if(c_population < 10000000) { return "City"; }
        else return "MegaCity";
    }

    int City::getPopulation() { return c_population; }

    float City::getSize() { return c_size; }

    std::string City::getName() { return c_name; }

    void City::setSize(float area) {
        if(area <= 0) { throw std::invalid_argument("Error, Not Enough Land!"); }
        else c_size = area;

    }
}